docs/rules/order.md

```diff
@@ -285,6 +285,49 @@ import index from './';
 import sibling from './foo';
 ```
 
+### `named: true|false|{enabled: true|false, import: true|false, export: true|false, require: true|false, groups: [array]}`
+
+Enforce ordering of names within imports:
+
+ - If set to `true`, named imports must be ordered according to the `alphabetize` options
+ - If set to `false`, named imports can occur in any order
+
+`import`, `export` and `require` allows forcing ordering of named `import`s/`export`s for the corresponding kind of expression.
+Use `enabled` to force ordering all of the aforementioned groups.
+
+The `groups` option allows to specify the order of the following kind of named `import`/`export` expressions:
+
+ - `value`: Importing/exporting ordinary values
+ - `default`: Exporting the `default` member in an expression like `export { default as Console } from './my-awesome-console.js'`
+ - `type`: Explicitly importing or exporting types: `import { type Readline } from 'node:readline'`
+
+The default value is `false` and the default `groups`-option is `['default']`.
+
```

Comment:
can you help me understand the use case for wanting name order to only be enforced in specific groups, or in different orders inside the same file?

Reply:
I probably didn't do much of a great job explaining this option.
This option allows to specify the order of the corresponding groups of specifiers.

So, the default `['default']` expresses "`default` first, everything else second" while `[['default', 'value'], 'type']` expresses "`default` and `value` specifiers first, `type` specifiers second" and `[]` expresses that every group has the same rank.

So this option allows specifying the order of the groups rather than whether or not they are enabled.

Reply:
ok so specifically, this is about determine the relative order of "default" vs "non-default values" vs "type imports"?

note that type imports can be a default as well (`import type foo from 'bar'`), and `import { default }` is a "named" import of the default, so this might get a bit complex.

In general, I think the simple thing to do here is support an option for types vs values, the choices being "mixed", "types first", or "types last". Would that be sufficient, do you think?

Reply:
The `import type foo from 'bar'` would be untouched by the `named`-ordering, as the new logic only affects the named specifiers part (the `{ A, B, C }` part in `import foo, { A, B, C } from 'bar').
I think having the choices `mixed`, `types first` and `types last` would be more simple to use. I thought it might be nice to give the users full control, though.

If you think the `mixed`, `types first/last` option is more suitable, I will work on changing the code accordingly.

Reply:
ahhh. if we're only talking about inside the curly braces, i don't think "default" should get any special treatment. If you want it to follow the semantics of being a default, then you use it in default position and not as a name.

Reply:
Oh - that's true, though, yes
I think I will drop the special treatment of `default` out, then!